extensions/amp-story-subscriptions/0.1/amp-story-subscriptions.js

```diff
@@ -2,30 +2,82 @@ import * as Preact from '#core/dom/jsx';
 import {Layout_Enum} from '#core/dom/layout';
 
 import {CSS} from '../../../build/amp-story-subscriptions-0.1.css';
+import {
+  StateProperty,
+  getStoreService,
+} from '../../amp-story/1.0/amp-story-store-service';
 
 const TAG = 'amp-story-subscriptions';
 
+/**
+ * The index of the limited-content page, which is the page where the paywall would be triggered.
+ * @const {number}
+ */
+const FIRST_PAYWALL_STORY_PAGE_INDEX = 2;
+
 export class AmpStorySubscriptions extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(this.win);
   }
 
   /** @override */
   buildCallback() {
+    // Mark pages with required attributes to be treated as paywall protected pages.
+    Array.prototype.map.call(
```

Comment:
We can use `iterateCursor` here https://github.com/ampproject/amphtml/blob/23e5f09e14f224fb57a473565e9ec4da4bc99aa4/src/core/dom/index.js#L320

Reply:
Done. But why is it better? Can I make the assumption that the AMP repo's own functions are always better that we optimize them for AMP usage?

Reply:
In general using the helpers are better for 2 reasons: 
1. Minification - our compiler can find all built in methods which cannot be minified and replace them with a single function like `a()`
2. Optimization - we can lean on our teammates to explore the best possible way to do something, and apply it to many use cases. `iterateCursor` is a good example here, someone took the time to figure out the most performant way to do this https://jsbench.github.io/#f638cacc866a1b2d6e517e6cfa900d6b and we can just take advantage of that. If someone ever decides to further optimize this function we would get that bonus without having to change any of this extension code.

Reply:
Wow that link is interesting. Yes it makes sense that people on AMP are always more actively trying to optimizing functions than the general JS dev community. Thanks for the info, Caleb!